NUService.js

```diff
@@ -333,21 +333,16 @@ export default class NUService extends NUObject {
      * Create a copy of NUService
      */
     clone() {
-        const newService = new NUService();
-        newService._APIKey = this._APIKey;
-        newService._headerAuthorization = this._headerAuthorization;
-        newService._headerCount = this._headerCount;
-        newService._headerFilter = this._headerFilter;
-        newService._headerFilterType = this._headerFilterType;
-        newService._headerMessage = this._headerMessage;
-        newService._headerOrderBy = this._headerOrderBy;
-        newService._headerPage = this._headerPage;
-        newService._headerPageSize = this._headerPageSize;
-        newService._password = this._password;
-        newService._rootURL = this._rootURL;
-        newService._userName = this._userName;
-        newService._pageSize = this._pageSize;
-        newService._customHeaders = this._customHeaders;
-        return newService;
+        const newService = new NUService({rootURL: this.rootURL});
+        return Object.assign(newService, this);
```

Comment:
this doesn't work, headers is an object.
as per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign, "For deep cloning, we need to use other alternatives because Object.assign() copies property values. If the source value is a reference to an object, it only copies that reference value"